src/core/Table/TablePaginator.tsx

```diff
@@ -177,6 +186,13 @@ export const TablePaginator = (props: TablePaginatorProps) => {
   );
   const [overflowRef, visibleCount] = useOverflow(pageList);
 
+  const paginatorRef = React.useRef<HTMLDivElement>(null);
+  const [currentWidth, setCurrentWidth] = React.useState(
+    () => paginatorRef.current?.offsetWidth ?? 0,
+  );
+  const [resizeRef] = useResizeObserver(({ width }) => setCurrentWidth(width));
+  const paginatorRefs = useMergedRefs(paginatorRef, resizeRef);
```

Comment:
This is actually buggy, which is why the test is failing. Once #436 is merged, I can just reuse the logic from there